kedro-airflow/kedro_airflow/grouping.py

```diff
@@ -25,74 +20,88 @@ def get_memory_datasets(catalog: DataCatalog, pipeline: Pipeline) -> set[str]:
     }
 
 
-def node_sequence_name(node_sequence: list[Node]) -> str:
-    return "_".join([node.name for node in node_sequence])
+def _build_adjacency_matrix(
+    catalog: DataCatalog, pipeline: Pipeline
+) -> tuple[dict[str, set], dict[str, set]]:
+    """
+    Builds adjacency matrix (adj_matrix) to search connected components - undirected graph,
+    and adjacency matrix (parent_to_children) to retrieve connections between new components using on
+    initial kedro topological sort - directed graph.
+    """
+    memory_datasets = get_memory_datasets(catalog, pipeline)
+
+    adj_matrix: dict[str, set] = {node.name: set() for node in pipeline.nodes}
```

Comment:
Is this really an adjacency matrix? I was thinking https://en.wikipedia.org/wiki/Adjacency_matrix but it seems like it's just another dictionary

Reply:
It's an adjacency list: https://en.wikipedia.org/wiki/Adjacency_list. Will rename it

Reply:
Done